### PR TITLE
feat(loader): adds a loader with overlay option, tweaks SB and handling of .type.ts files

### LINKED ISSUE #23: Loading spinner
Simple CSS loading spinner, to be displayed while data updates are happening - will need to have a short delay built in so it only shows when needed (e.g. only if a request takes longer than half a second).

### PR DESCRIPTION
**Pull request by xanthipe**
**Issues affected**

Closes #23 

**Description**

* Adds a loader component with an overlay option.
* Tweaks Jest settings to run on watch, not on save.
* Updates SB which appears to have helped with stability issues.

**How to test**

Run up SB, poke the Display/Loader stories.

**Screenshots**

Not the best because it's an animated loader, but...

image

**Any other information?**

This is very much a v1, I expect this to change as we go...

**Checklist**

- [x] Have all new components been added to Storybook?
- [ ] Have unit/component/integration tests been written?
- [ ] Has documentation been updated?

### CODE REVIEW

File: src/components/display/icon/icon.types.ts [hunk 1]
```diff
@@ -1,28 +1,27 @@
...
-import { ReactComponent as Rank } from '../../assets/icons/stock-line.svg';
-import { ReactComponent as Gm } from '../../assets/icons/survey-line.svg';
-import { ReactComponent as Game } from '../../assets/icons/sword-fill.svg';
-import { ReactComponent as Twitter } from '../../assets/icons/twitter-fill.svg';
-import { ReactComponent as AttendanceSignUp } from '../../assets/icons/user-follow-fill.svg';
-import { ReactComponent as Attending } from '../../assets/icons/user-line.svg';
-import { ReactComponent as NotAttending } from '../../assets/icons/user-unfollow-line.svg';
-import { ReactComponent as Committee } from '../../assets/icons/vip-crown-line.svg';
-import { ColorTypes } from '../../themes/theme.types';
+import { ReactComponent as Account } from '../../../assets/icons/account-circle-line.svg';
```
[the remark this answers is not in the document]
Comment (xanthipe) [reply src/components/display/icon/icon.types.ts:3]: We could, but do we really need to if they're just going to be in this component? You shouldn't need to import them anywhere else?

[the remark this answers is not in the document]
Comment (DiracLotus) [reply src/components/display/icon/icon.types.ts:3]: if we're happy we won't be needing to store any images or other stuff in there, then sure - works for me :)

[the remark this answers is not in the document]
Comment (DiracLotus) [reply src/components/display/icon/icon.types.ts:3]: mostly thinking about non-icon assets

[the remark this answers is not in the document]
Comment (xanthipe) [reply src/components/display/icon/icon.types.ts:3]: Off the top of my head, the only other image we're going to need is the header logo and favicon...can we come back to this if it turns out to be annoying later?

**Pull request merged**